### DISCUSSION TITLE
Animation repetitions do not work when specifiying animation name

### DISCUSSION THREAD

**Discussion by LazZiya** [google/model-viewer#3149] [Q&A] [answered]
The repetitions count is not respected when specifiying animation name.

````js
// This will keep looping
mv.animationName = "other-animation";
mv.play({ repetitions: 1 });
````

Without  `animationName` it will work fine for the first animaton only.

#### Live Demo
https://glitch.com/edit/#!/chatter-pumped-armadillo

### Version
- model-viewer: v1.10.1

### Browser Affected

- [x] Chrome, version: 97.0.4692.99
- [x] Edge
- [x] Safari

### OS

- [x] Windows
- [x] iOS

**Comment by futahei** [comment, marked as answer]
This may be a temporary workaround, but it will work correctly if you do the following this,
```
mv.animationName = "Death";
await mv.updateComplete;
mv.play({repetitions: 1});
```
Information about `updateComplete` is [here](https://lit.dev/docs/v1/components/lifecycle/#updatecomplete)

**Reply by elalish** [reply]
Indeed, I wouldn't even  call that a temporary workaround so much as just how Lit elements work (we inherit from their library). Everything is async, so `await mv.updateComplete`  is necessary for consecutive operations for which the order is important, particularly attributes. You'll notice it is quite pervasive  in our tests.

**Comment by LazZiya** [comment]
Thank you @futahei , the workaround worked fine 👍

**Comment by designbyadrian** [comment]
I experienced the same issue with version 1.9.2. Updating to version 1.10.1 solves the issue if I apply the fix above.

**Comment by futahei** [comment]
@designbyadrian This feature is available since v1.10.0.
Please refer to the release notes [v1.10.0](https://github.com/google/model-viewer/releases/tag/v1.10.0).
